Kernel/Bus/USB/UHCI/UHCIController.cpp

```diff
@@ -510,6 +519,41 @@ ErrorOr<size_t> UHCIController::submit_bulk_transfer(Transfer& transfer)
     return transfer_size;
 }
 
+ErrorOr<void> UHCIController::submit_async_interrupt_transfer(NonnullLockRefPtr<Transfer> transfer, u16 ms_interval)
+{
+    dbgln_if(UHCI_DEBUG, "UHCI: Received interrupt transfer for address {}. Root Hub is at address {}.", transfer->pipe().device_address(), m_root_hub->device_address());
+
+    if (ms_interval == 0) {
+        return EINVAL;
+    }
+
+    SpinlockLocker locker(m_async_lock);
```

Comment:
this lock makes me nervous. You are doing multiple allocations here with it held. Kmalloc with a spinlock held is muy no bueno. at least move it below the create_transfer_queue call.